ksqldb-engine/src/main/java/io/confluent/ksql/execution/pull/StreamedRowTranslator.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.pull;
+
+import com.google.common.base.Preconditions;
+import io.confluent.ksql.rest.entity.KsqlHostInfoEntity;
+import io.confluent.ksql.rest.entity.StreamedRow;
+import io.confluent.ksql.rest.entity.StreamedRow.Header;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.util.ConsistencyOffsetVector;
+import io.confluent.ksql.util.KsqlException;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+public class StreamedRowTranslator implements Iterator<PullQueryRow> {
+
+  private final Deque<PullQueryRow> rowQueue = new ArrayDeque<>();
+  private final LogicalSchema expectedSchema;
+  private final Optional<ConsistencyOffsetVector> offsetVector;
+
+  private Header header = null;
+
+  public StreamedRowTranslator(
+      final LogicalSchema expectedSchema,
+      final Optional<ConsistencyOffsetVector> offsetVector
+  ) {
+    this.expectedSchema = expectedSchema;
+    this.offsetVector = offsetVector;
+  }
+
+  @Override
+  public PullQueryRow next() {
+    return rowQueue.pollFirst();
+  }
+
+  @Override
+  public boolean hasNext() {
+    return !rowQueue.isEmpty();
+  }
```

Comment:
Hmm... I don't think that the semantics passed through from `rowQueue` do not match the `Iterator` interface.

That makes me a little nervous...

Reply:
heh fair, I'll just remove the `implements Iterator` part of it. It's really just a queue we drain every time. I can call it that.